components/director/internal/domain/apptemplate/service.go

```diff
@@ -321,7 +321,14 @@ func (s *service) PrepareApplicationCreateInputJSON(appTemplate *model.Applicati
 	appCreateInputJSON := appTemplate.ApplicationInputJSON
 	for _, placeholder := range appTemplate.Placeholders {
 		newValue, err := values.FindPlaceholderValue(placeholder.Name)
-		if err != nil {
+		var optional bool
+		if placeholder.Optional == nil {
+			optional = false
+		} else {
+			optional = *placeholder.Optional
+		}
+
+		if err != nil && !optional {
```

Comment:
This can be simplified:
```suggestion
		isOptional := false
		if placeholder.Optional != nil {
			isOptional = *placeholder.Optional
		}

		if err != nil && !isOptional {
```

What do you think?

Reply:
Yes, it looks better when simplified. I will refactor it in another PR so that this one can be released.